### PR TITLE
Address inconsistency between SelectOne widgets ,fix issue #1855

### LINKED ISSUE #1855: Inconsistency between SelectOne widgets.
#### Software and hardware versions 
Current master branch

#### Problem description
Last month we implemented `image-map` appearance for select questions. I noticed a small inconsistency between that new widget and ordinary `SelectOneWidget`. 
If we use the first one we can unselect our answer clicking on the same value what is not possible in `SelectOneWidget`. 

#### Expected behavior
I'm not sure which approach we should choose but it should work in the same way. @lognaturel what do you think?

#### Other information 
singleselect

Form for testing:
[canada.zip](https://github.com/opendatakit/collect/files/1719434/canada.zip)

@opendatakit-bot label "needs discussion"
@opendatakit-bot label "quick win"

### PR DESCRIPTION
**Pull request by huangyz0918**
Closes #1855 

#### What has been done to verify that this works as intended?

I have readed others' discuss about this problem and I think it's easy to achieve this goal, so I check the `SelectOneImageMapWidget.java` which calls the `svg _map_helper.js` by an internal Java interface `@android.webkit.JavascriptInterface` and founded that the only way to address this issue is to modify the Javascript file which called by Java methods.

So I modified the function : `clickOnArea(areaId)` in `svg _map_helper.js` which is related to the twice-click action ( you can check codes in my branch `issue#1855` )

Here you can see the demos:

Single-selection mode:
single mode

Multiple-selections mode:
multiple mode

#### Why is this the best possible solution? Were any other approaches considered?

We use a Android WebView to display the map and it related to the JavaScript injection in Android, so it's likely the only way to address this issue is changing logic inside the Javascript file because of the isolation between Android package and js.

I think it may exist other approaches to meet these requirtments but it maybe more complex than this solution.

#### Are there any risks to merging this code? If so, what are they?

From the tests I have done before submiting pull request, I think the new project has no risks than previous work:

1. I have only modified the logic inside `svg _map_helper.js` ,with the same `JavascriptInterface` that java methods called, it won't impact the logic of Java.

2. I considered either `Single-option Mode` and `Multiple-options Mode`, they all have past the testing and work well.

#### Do we need any specific form for testing your changes? If so, please attach one.

I think you don't have to test more, just review the code and run on several devices .
I think it's okey if you are willing to take futher testing :)

### CODE REVIEW

File: collect_app/src/main/assets/svg_map_helper.js [hunk 1]
```diff
@@ -61,20 +61,28 @@ function setSelectMode(isSingleSelect) {
...
+            selectedAreas.push(areaId);
+            selectArea(areaId);
+            lastSelectedAreaId = areaId;
+        }else{
+             // multiple select mode
+            if (selectedAreas.indexOf(areaId) !== -1) {
+                document.getElementById(areaId).setAttribute('style', 'fill: ' + originalColors[areaId]);
+                selectedAreas.splice(selectedAreas.indexOf(areaId), 1);
+                unselectArea(areaId);
+            }else{
```
Comment (grzesiek2010) [inline collect_app/src/main/assets/svg_map_helper.js]: space before and after `else`

File: collect_app/src/main/assets/svg_map_helper.js [hunk 2]
```diff
@@ -61,20 +61,28 @@ function setSelectMode(isSingleSelect) {
...
+            if (Boolean(isSingleSelect) && !!lastSelectedAreaId) {
+                document.getElementById(lastSelectedAreaId).setAttribute('style', 'fill: ' + originalColors[lastSelectedAreaId]);
+                selectedAreas.splice(selectedAreas.indexOf(lastSelectedAreaId), 1);
+                unselectArea(lastSelectedAreaId);
+            }
+            document.getElementById(areaId).setAttribute('style', 'fill: #E65100');
+            selectedAreas.push(areaId);
+            selectArea(areaId);
+            lastSelectedAreaId = areaId;
+        }else{
```
Comment (grzesiek2010) [inline collect_app/src/main/assets/svg_map_helper.js]: space before and after `else`

File: collect_app/src/main/assets/svg_map_helper.js [hunk 3]
```diff
@@ -61,20 +61,28 @@ function setSelectMode(isSingleSelect) {
...
-    if (selectedAreas.indexOf(areaId) !== -1) {
-        document.getElementById(areaId).setAttribute('style', 'fill: ' + originalColors[areaId]);
-        selectedAreas.splice(selectedAreas.indexOf(areaId), 1);
-        unselectArea(areaId);
-    } else {
-        if (Boolean(isSingleSelect) && !!lastSelectedAreaId) {
-            document.getElementById(lastSelectedAreaId).setAttribute('style', 'fill: ' + originalColors[lastSelectedAreaId]);
-            selectedAreas.splice(selectedAreas.indexOf(lastSelectedAreaId), 1);
-            unselectArea(lastSelectedAreaId);
+        if (Boolean(isSingleSelect)){
```
Comment (grzesiek2010) [inline collect_app/src/main/assets/svg_map_helper.js]: space before `{`

**Review by grzesiek2010 — CHANGES_REQUESTED** [review]
It looks ok but please clean the code and your branch because you have unrelated changes here.

**Comment by huangyz0918** [comment]
@grzesiek2010 okey 
I will format the js code right now

**Review by grzesiek2010 — APPROVED** [review]

**Comment by grzesiek2010** [comment]
@opendatakit-bot label "needs testing"

**Comment by huangyz0918** [comment]
@grzesiek2010 I think I got sufficient testing already
do you suggest I need do more testing ? that I am not sure about that
I'm new and please advise ,thanks : )

**Comment by grzesiek2010** [comment]
`needs testing` means it's ready to test by our QAs.

**Comment by mmarciniak90** [comment]
**Tested with success!**

Verified on Android: 4.1, 4.2, 4.4, 5.1, 6.0, 7.0, 8.0

Verified cases: 
* Image select one widget
* Image select multiple widget
* Other select one widgets available in `All widgets` form
* Other select multiple widgets available in `All widgets` form
* New and saved form

@opendatakit-bot unlabel "needs testing"
@opendatakit-bot label "behavior verified"

**Review by shobhitagarwal1612 — APPROVED** [review]
@huangyz0918 Welcome to ODK and congrats on your first PR! :100:

**Pull request merged**